css/css-text/letter-spacing/letter-spacing-bidi-003.xht

```diff
@@ -0,0 +1,48 @@
+﻿<html xmlns="http://www.w3.org/1999/xhtml">
+  <head>
+    <meta charset="utf-8"/>
+    <title>CSS Text level 3 Test: letter spacing, justified text and bidi</title>
+    <link rel="author" title="Mike Bremford" href="http://bfo.com" />
+    <link rel="help" href="https://drafts.csswg.org/css-text-3/#letter-spacing-property" />
+    <link rel="help" href="https://drafts.csswg.org/css-writing-modes-4/#bidi-fragmentation" />
+    <link rel="match" href="reference/letter-spacing-bidi-003-ref.xht" />
+    <meta name="assert" content="text-align: justify will stretch the space between words, which should be applied after bidi processing."/>
+    <style>
+      .test, .control {
+          font: 50px/1 monospace;
+          margin: 0;
+          width: 14ch;
+          font-kerning: none;
+          border: 1px solid black;
+      }
+      .control {
+          position: absolute;
+          color: red;
+          z-index: -1;
+          white-space: pre;
+      }
+      .rtl {
+          direction: rtl;
+      }
+      .justify {
+          text-align-last: justify;
+      }
+      span {
+          background-color: #eee;
+      }
+      .narrow {
+          letter-spacing: 1ch;
+      }
+    </style>
+ </head>
+ <body>
+  <p>Test passes if no red is visible.</p>
+  <!-- bet <ls> alef <ls> <space stretched from justification> A <ls> B -->
+  <div class="control">&#x5d0; &#x5d1;<div style="float: right">A B</div></div>
```

Comment:
Unfortunately, minutiae of text rendering means that black text on top of identical red text will not reliably hide the red, and some small portion of it may show through due to anti-aliasing. For "A B" this could be worked around using the Ahem font, but for "&#x5d0; &#x5d1;" that doesn't help.

Maybe you could instead put a span around each letter in the "control" div, add something like `.control span {background:orange; color: transparent;}` and change the sentence from "passes if no red is visible" to "passes if all the letters are inside the orange boxes". Or you could keep the test as-is, and add the black-on-red in the reference as well. Feel free to come up with something else, but I think it would be good to avoid something that can fail due to painting artifacts even if the layout is correct.

Reply:
Very true. Given I'm already testing with a background in some of these, I've gone with red-text behind the black text for the reference tests, and noted that anti-aliasing doesn't matter in the descriptive text.

Ideally we'd have a RTL character in Ahem of course - I'm certainly happy to add this, but it's a very far-reaching change.